superset/db_engine_specs/bigquery.py

```diff
@@ -213,6 +218,52 @@ def fetch_data(
             data = [r.values() for r in data]  # type: ignore
         return data
 
+    @classmethod
+    def _reformat_error_message(cls, message: str) -> ErrorMessageType:
+        """Reformat error message for user experience"""
+        splitted_message = re.split("\n+", message, 1)
+        if len(splitted_message) > 1:
+            return ErrorMessageType(
+                short_message=splitted_message[0], description=splitted_message[1]
+            )
+
+        return ErrorMessageType(short_message=splitted_message[0], description="")
+
+    @classmethod
+    def extract_errors(
```

Comment:
Feels like this should be added to `superset/db_engine_specs/base.py`, and called somewhere (?) It's hard to review this as is without the larger context of how it's handled for other engines, and where it's called